mk/config.mk

```diff
@@ -226,3 +226,6 @@ ifneq ($(CFG_TA_GPROF_SUPPORT),y)
 $(error Cannot instrument user libraries if user mode profiling is disabled)
 endif
 endif
+
+# Enable Secure Data Path support in OP-TEE core
```

Comment:
I'd like to see some short explanations here (i.e., what happens if I enable this flag?). For instance the comments you have added to `check_params()` ("When CFG_WITH_SDP is [not] enabled...") would fit here nicely I think.
It should be clear than when this is enabled, OP_TEE knows about a physical range of secure memory (CFG_TEE_SDP_MEM_BASE, CFG_TEE_SDP_MEM_SIZE) that can be used to pass memrefs between NW and TAs, and that TA access rules are as you said in your comments.